subprojects/testing-jvm/src/main/java/org/gradle/api/internal/tasks/testing/detection/DefaultTestClassScanner.java

```diff
@@ -33,12 +35,14 @@
     private final FileTree candidateClassFiles;
     private final TestFrameworkDetector testFrameworkDetector;
     private final TestClassProcessor testClassProcessor;
+    private final TestClassSelectionMatcher testClassSelectionMatcher;
 
     public DefaultTestClassScanner(FileTree candidateClassFiles, TestFrameworkDetector testFrameworkDetector,
-                                   TestClassProcessor testClassProcessor) {
+                                   TestClassProcessor testClassProcessor, DefaultTestFilter testFilter) {
```

Comment:
❌ Please don't make the scanner more complicated. Use a decorating test class processor instead.